BonusCalcApi/Startup.cs

```diff
@@ -132,11 +132,19 @@ private void ConfigureDbContext(IServiceCollection services)
             var connectionString = Environment.GetEnvironmentVariable("CONNECTION_STRING")
                                 ?? Configuration.GetValue<string>("DatabaseConnectionString");
 
-            services.AddDbContext<BonusCalcContext>(
-                opt => opt
-                    .UseNpgsql(connectionString)
-                    .UseSnakeCaseNamingConvention()
-            );
+            services.AddDbContext<BonusCalcContext>(options =>
+            {
+                var builder = new NpgsqlDataSourceBuilder(connectionString);
+                builder.MapEnum<BandChangeDecision>("band_change_decision");
+                var dataSource = builder.Build();
```

Comment:
I don't follow this.
From what I understand before this change, it line135 says: if `BonusCalcContext` is needed, then `.AddDbContext` ensures this class's creation with the specified options delegate.

Looking at the new changes, all of that stays the same, so what does the `dataSource` change?
Is it mandatory to use the `UseNpgsql` with the newer package versions?
Had you not specified these 3 lines (137-139), would you get some kind of error?

Reply:
Having read through one of the commits:
I'm guessing this block is needed solely to configure this extra logic for enum mapping.